src/components/uri/Visualization.tsx

```diff
@@ -667,11 +667,26 @@ export const IntlVisualization = injectIntl(VisualizationWrapped);
  * [Visualization](http://sdk.gooddata.com/gooddata-ui/docs/react_components.html#visualization)
  * is a component that renders saved visualization based on projectId and either identifier or uri
  */
-export class Visualization extends React.PureComponent<IVisualizationProps> {
+export class Visualization extends React.PureComponent<IVisualizationProps, { locale: "en-US" }> {
+    constructor(props: IVisualizationProps) {
+        super(props);
+
+        this.state = {
+            locale: "en-US",
+        };
+    }
+
+    public async componentDidMount() {
+        this.setState({ locale: (await sdk.user.getCurrentProfile()).language });
```

Comment:
What if the sdk call fails?
What if it takes several seconds? Will it show in English first and then change?

Reply:
If it takes longer than the loading of the Visualization, it will default to English and then change. However, its very unlikely because I think the Visualization will take longer than pulling information from the user and using it.

As for when the sdk call fails, I am not sure if it will throw an error. Could you give me a tip on how to make sdk call fail?

Reply:
Good point, however it can happen because the results are cached, so reloading the page the visualization can be very fast.

It will definitely throw an error. To quickly test the fail, replace the `sdk.user.getCurrentProfile()` with `() => Promise.reject(new Error("Message"))`.